components/ContributionConfirmationModal.js

```diff
@@ -0,0 +1,237 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { getCurrencySymbol } from '../lib/currency-utils';
+import { i18nGraphqlException } from '../lib/errors';
+import { API_V2_CONTEXT, gqlV2 } from '../lib/graphql/helpers';
+
+import Container from './Container';
+import FormattedMoneyAmount from './FormattedMoneyAmount';
+import { Box, Flex } from './Grid';
+import InputField from './InputField';
+import StyledButton from './StyledButton';
+import StyledHr from './StyledHr';
+import Modal, { ModalBody, ModalFooter, ModalHeader } from './StyledModal';
+import { P, Span } from './Text';
+import { TOAST_TYPE, useToasts } from './ToastProvider';
+
+const confirmContributionMutation = gqlV2/* GraphQL */ `
+  mutation ConfirmContribution($id: String!, $action: ProcessOrderAction!, $details: OrderDetailsInput) {
+    processPendingOrder(order: { id: $id }, action: $action, details: $details) {
+      id
+      status
+      permissions {
+        id
+        canMarkAsPaid
+        canMarkAsExpired
+      }
+      amount {
+        currency
+        valueInCents
+      }
+      platformContributionAmount {
+        currency
+        valueInCents
+      }
+    }
+  }
+`;
+
+const ContributionConfirmationModal = ({ order, onClose, show }) => {
+  const amountInitiated = order.amount.valueInCents + order.platformContributionAmount.valueInCents;
+  const currency = order.amount.currency;
+  const [amountReceived, setAmountReceived] = useState(amountInitiated);
+  const [platformTip, setPlatformTip] = useState(order.platformContributionAmount.valueInCents);
+  const [paymentProcessorFee, setPaymentProcessorFee] = useState(0);
+  const [netAmount, setNetAmount] = useState(amountReceived - platformTip);
+  const intl = useIntl();
+  const { addToast } = useToasts();
+  const [confirmOrder, { loading }] = useMutation(confirmContributionMutation, { context: API_V2_CONTEXT });
+
+  useEffect(() => {
+    setNetAmount(amountReceived - platformTip);
+  }, [amountReceived, platformTip]);
```

Comment:
Why use a state + effect for this? Why not just:

```suggestion
  const netAmount = amountReceived - platformTip;
```

Reply:
Yes correct, we could simplify. 👍🏽